sam_view.c

```diff
@@ -795,50 +858,215 @@ static bool bam1_to_fq(const bam1_t *b, kstring_t *linebuf, const bam2fq_state_t
     return true;
 }
 
+/*
+ * Create FASTQ lines from the barcode tag using the index-format 
+ */
+static bool tags2fq(bam1_t *rec, const bam2fq_state_t *state, const bam2fq_opts_t* opts)
+{
+    uint8_t *p;
+    char *ifmt_orig = strdup(opts->index_format);
```

Comment:
Hmmm. Historical reasons. I think that getLength(&ifmt) *did* change the data at one time.